### PR TITLE
Simplified aggregated DTOs

### LINKED ISSUE #98: RaceParticipation REST [deadline: 19.12]
- [x] Fixes `RaceParticipation` POST request. It should receive `ids` of entities, not the whole DTO objects.
- [x] `RaceParticipation` PUT. It should allow updating only result position or ids of relations.
- [x] Changes in `Driver`, `Race` or `CarSetup` endpoints should be reflected in `RaceParticipation` GET.

### PR DESCRIPTION
**Pull request by elderanakain**
Fixes #98, #99 

I have decided to use ids in DTOs instead of the full object. It is not the perfect variant, but it was easier and faster to do. Also, having DTO objects as fields caused stack overflow during mapping Entity -> DTO.

cc: @Adel83, @Swaglis

### CODE REVIEW

File: formulaoneteam-web/src/main/webapp/WEB-INF/jsp/drivers/list.jsp [hunk 1]
```diff
@@ -37,7 +37,14 @@
...
                             <c:forEach items="${driver.raceParticipations}" var="participation">
-                                <my:a href="/world-championship/detail/${participation.id}">${participation.race.title}</my:a><br/>
+                                <my:a href="/world-championship/detail/${participation.id}">
+                                    <c:forEach var="race" items="${races}">
+                                        <c:if test="${race.key == participation.raceId}">
+                                            ${race.value.title}
+                                        </c:if>
+                                    </c:forEach>
+                                    <br/>
+                                </my:a>
```
Comment (mrnustik) [inline formulaoneteam-web/src/main/webapp/WEB-INF/jsp/drivers/list.jsp]: You also have to do this on Driver detail page where also all his races and test drives are accessed.

File: formulaoneteam-web/src/main/java/cz/muni/fi/pa165/mvc/controllers/DriversController.java [hunk 2]
```diff
@@ -26,13 +29,19 @@
...
-    ;
 
     @RequestMapping("/list")
     public String list(Model model) {
-        model.addAttribute("drivers", driverFacade.getAllDrivers());
+        model.addAttribute("drivers", driverFacade.getAll());
+        model.addAttribute(
+                "races",
+                raceFacade.getAll().stream().collect(Collectors.toMap(BaseDTO::getId, race -> race))
+        );
```
Comment (mrnustik) [inline formulaoneteam-web/src/main/java/cz/muni/fi/pa165/mvc/controllers/DriversController.java]: You also have to do this on Driver detail page where also all his races and test drives are accessed.

**Comment by elderanakain** [comment]
@mrnustik I have changed the flow.
Yeah, you got the right point, thanks!

**Comment by elderanakain** [comment]
@mrnustik Could you check it? Cause I am blocked by this PR.

**Comment by mrnustik** [comment]
Approved. Sorry for taking that long.

**Pull request merged**